### PR TITLE
containerized dag deployment code

### PR DESCRIPTION
**Pull request by abhisheksawalkar2001**

### CODE REVIEW

File: src/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,6 @@
+FROM python
+
+WORKDIR /code
+COPY . /code
+
```
Comment (santoshgopane) [inline src/Dockerfile:5]: add upgrade pip piece here

File: src/app/common/config.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import os
```
Comment (santoshgopane) [inline src/app/common/config.py:1]: either segregate them by class or functions

File: src/app/main.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        """This function preprocessing on downloaded s3 bucket file in pandas dataframe"""
+        df[["fname", "lname"]] = df.name.str.split(" ", expand=True)
+
+        df = df[["fname", "lname", "age"]]
+
+        return df
+
+    def extract_api_data(self):
+        """This function grabing Hearthstone API data through RapidAPI Hub"""
+        conn = http.client.HTTPSConnection(host=config.api_host)
```
Comment (santoshgopane) [inline src/app/main.py:43]: use requests module of python for such tasks

File: src/app/main.py [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+        data = json.loads(data.decode("utf-8"))["Basic"]
+
+        return data
+
+    def transform_api_data(self, data):
+        """This function preprocessing hearthstone API data of cards"""
+        df = pd.json_normalize(data)
+
+        def prep(row):
```
Comment (santoshgopane) [inline src/app/main.py:63]: You should not use methods to tackle, this looks like to handle a single case. But you should think of solution that is applying to all data.

File: src/app/main.py [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+
+        df["mechanics"] = df["mechanics"].apply(prep)
+
+        df.rename(columns={"mechanics": "mechanics_name"})
+
+        return df
+
+    def load(self, df, table_name):
+        """This is a connection with MySQL database and writing dataframe into table in the databases"""
+        engine = create_engine(
```
Comment (santoshgopane) [inline src/app/main.py:77]: add into config.py

File: src/app/main.py [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+                user=config.mysql_user,
+                pw=config.mysql_password,
+                db=config.mysql_database,
+            )
+        )
+
+        df.to_sql(con=engine, name=table_name, if_exists="append", index=False)
+
+
+if __name__ == "__main__":
```
Comment (santoshgopane) [inline src/app/main.py:89]: dont have this here instead create another python file that creates class object

File: src/requirements.txt [hunk 7]
```diff
@@ -0,0 +1,4 @@
+pandas
```
Comment (santoshgopane) [inline src/requirements.txt:1]: have specific version of each package.

**Review by santoshgopane — CHANGES_REQUESTED** [review]
Please work on the changes and resubmit

File: src/app/common/config.py (hunk 2 above)
Comment (abhisheksawalkar2001) [reply src/app/common/config.py:1]: means all cred in classes and functions bind

Comment (santoshgopane) [reply src/app/common/config.py:1]: yes

File: src/app/main.py (hunk 6 above)
Comment (abhisheksawalkar2001) [reply src/app/main.py:89]: can i remove this line and direct create a class object and call functions

**Pull request merged**